sonar-core/src/main/java/org/sonar/core/util/DefaultHttpDownloader.java

```diff
@@ -75,7 +75,7 @@ public DefaultHttpDownloader(Server server, Settings settings, @Nullable Integer
   public DefaultHttpDownloader(Server server, Settings settings, @Nullable Integer connectTimeout, @Nullable Integer readTimeout) {
     this.readTimeout = readTimeout;
     this.connectTimeout = connectTimeout;
-    downloader = new BaseHttpDownloader(settings.getProperties(), server.getVersion());
+    downloader = new BaseHttpDownloader(new ProxySystem(), settings, server.getVersion());
```

Comment:
this pattern of `new ProxySystem()` and settings as arguments of the `BaseHttpDownloader`'s constructor caught my eye

it looks like it's only justified by the UT code testing how `ProxySystem` is populated from the `Settings`. This object is only built from `Settings`.

My advice, move `ProxySystem` out of `DefaultHttpDownloader` as a package protected class, make the constructor private and create a static method to create a `ProxySystem` instance from a `Settings` instance. Test the built object in the UT of `ProxySystem`. You'll end up with a `BaseHttpDownloader(ProxySystem system, @Nullable String userAgent)` constructor and better SOC.